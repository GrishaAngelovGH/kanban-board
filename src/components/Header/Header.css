.json-export-btn > .btn {
  font-size: 24px;
  color: rgb(108, 117, 125);
  padding: 5px;
  border: none;
  font-family: initial;
}

.json-export-btn > .btn:hover {
  color: rgb(108, 117, 125);
}
