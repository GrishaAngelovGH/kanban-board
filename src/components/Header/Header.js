import FilterMenu from "./DropdownMenus/FilterMenu"
import BoardMenu from "./DropdownMenus/BoardMenu"

import TaskProgress from "components/TaskProgress"

import boardRepository from "persistent/persistentKanbanBoardRepository"

const Header = ({ downloadJsonHref, disabledExport, handlers }) => {
  const { onUsersClick, onSettingsClick, onCalendarClick, onImportKanbanBoardClick, onUpdate } = handlers

  const columns = boardRepository.getColumns()

  return (
    <div className="row bg-secondary-subtle p-2 justify-content-between align-items-center">
      <div className="col-4 col-md-4 col-lg-2">
        <h3 className="m-0 text-center">Kanban Board</h3>
      </div>

      <div className="col-4 col-md">
        <TaskProgress
          className="m-0 p-md-2 p-lg-1 bg-transparent text-secondary"
          columns={columns}
        />
      </div>

      <div className="col-4 col-md-2 col-lg-1">
        <div className="d-flex justify-content-evenly">
          <FilterMenu onUpdate={onUpdate} />

          <BoardMenu
            downloadJsonHref={downloadJsonHref}
            disabledExport={disabledExport}
            onUsersClick={onUsersClick}
            onSettingsClick={onSettingsClick}
            onCalendarClick={onCalendarClick}
            onImportKanbanBoardClick={onImportKanbanBoardClick}
            onUpdate={onUpdate}
          />
        </div>
      </div>
    </div>
  )
}

export default Header