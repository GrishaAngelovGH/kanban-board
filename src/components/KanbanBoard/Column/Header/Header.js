import Button from "react-bootstrap/Button"
import ButtonGroup from "react-bootstrap/ButtonGroup"

import Menu from "../Menu"

import EditableText from "components/EditableText"
import NavigationButtons from "components/KanbanBoard/NavigationButtons"

const Header = ({
  id, title, limit, tasks, handlers,
  statuses: { markedAsDone, isGridView, isSingleRowView, hasNoBackground, hasSolidColumnStyle }
}) => {
  const handleAddTask = () => {
    handlers.onAddTask(id)
  }

  const handleMarkAsDone = () => {
    handlers.onMarkColumnAsDone(id)
  }

  const handleDeleteAction = () => {
    handlers.onDeleteColumn(id)
  }

  const handleDeleteAllTasks = () => {
    handlers.onDeleteAllTasks(id)
  }

  const handleShowLimit = () => {
    handlers.setShowLimit(true)
  }

  const handleAutoAssignUsers = () => {
    handlers.onAutoAssignUser(id)
  }

  const handleSwapToLeft = () => {
    handlers.onSwapColumn(id, true)
  }

  const handleSwapToRight = () => {
    handlers.onSwapColumn(id, false)
  }

  const titleClass = hasSolidColumnStyle || hasNoBackground ? "text-dark" : "text-white"
  const checkButtonClass = hasSolidColumnStyle ? "secondary-subtle" : "light"
  const taskLengthButtonClass = hasSolidColumnStyle ? "secondary" : "light"

  return (
    <div className="row">
      <div className={`${isGridView || isSingleRowView ? "col-7 col-lg-8" : "col-10"}`}>
        <NavigationButtons
          size="sm"
          className="mb-2"
          onLeftClick={handleSwapToLeft}
          onRightClick={handleSwapToRight}
        />

        <EditableText onBlur={handlers.handleUpdate}>
          <h3 className={`${titleClass} text-break text-capitalize`}>{title}</h3>
        </EditableText>

        <h3 className={titleClass}>{limit > 0 && `(${tasks.length} / ${limit})`}</h3>
      </div>
      <div className={`${isGridView || isSingleRowView ? "col-5 col-lg-4" : "col-2"}`}>
        <ButtonGroup size="sm">
          {markedAsDone && <Button variant={checkButtonClass} className="bi bi-check-circle-fill text-success" disabled />}

          <Button variant={taskLengthButtonClass} disabled>{tasks.length}</Button>

          <Menu
            markedAsDone={markedAsDone}
            handleAddTask={handleAddTask}
            handleMarkAsDone={handleMarkAsDone}
            handleDeleteAction={handleDeleteAction}
            handleDeleteAllTasks={handleDeleteAllTasks}
            handleShowLimit={handleShowLimit}
            handleAutoAssignUsers={handleAutoAssignUsers}
          />
        </ButtonGroup>
      </div>
    </div>
  )
}

export default Header