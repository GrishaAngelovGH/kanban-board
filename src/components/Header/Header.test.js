import { render, fireEvent, act, waitFor } from "@testing-library/react"

import Header from "./Header"

beforeEach(() => {
  const mockedDate = new Date(2023, 11, 22)
  jest.useFakeTimers("modern")
  jest.setSystemTime(mockedDate)
})

afterEach(() => {
  jest.useRealTimers()
})

test("should render Header component", async () => {
  const view = render(<Header />)

  const menu = view.container.querySelector('[role="group"]')

  act(() => {
    fireEvent.click(menu.firstChild)
  })

  await waitFor(() => {
    expect(view).toMatchSnapshot()
  })
})