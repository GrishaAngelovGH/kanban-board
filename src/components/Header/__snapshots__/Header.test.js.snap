// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row bg-secondary-subtle p-2 justify-content-between align-items-center"
      >
        <div
          class="col-4 col-md-4 col-lg-2"
        >
          <h3
            class="m-0 text-center"
          >
            Kanban Board
          </h3>
        </div>
        <div
          class="col-4 col-md"
        >
          <p
            class="bg-white rounded text-center p-1 m-0 p-md-2 p-lg-1 bg-transparent text-secondary"
          >
            Completed 
            0
             task
            s
             of 
            0
             (
            0
            %)
          </p>
        </div>
        <div
          class="col-4 col-md-2 col-lg-1"
        >
          <div
            class="d-flex justify-content-evenly"
          >
            <div
              class="dropdown-menu-header show dropdown btn-group"
              role="group"
            >
              <button
                aria-expanded="true"
                class="dropdown-toggle show btn btn-light btn-sm"
                type="button"
              >
                <i
                  class="bi bi-filter"
                />
              </button>
              <div
                aria-labelledby=""
                class="dropdown-menu show"
                x-placement="bottom-start"
              >
                <span
                  class="bg-secondary-subtle text-secondary border text-center dropdown-item-text"
                >
                  Filter by priority
                </span>
                <a
                  class="dropdown-item"
                  data-rr-ui-dropdown-item=""
                  href="#"
                  role="button"
                  tabindex="0"
                >
                  All
                   
                </a>
                <a
                  class="dropdown-item"
                  data-rr-ui-dropdown-item=""
                  href="#"
                  role="button"
                  tabindex="0"
                >
                  Low
                   
                </a>
                <a
                  class="dropdown-item"
                  data-rr-ui-dropdown-item=""
                  href="#"
                  role="button"
                  tabindex="0"
                >
                  Medium
                   
                </a>
                <a
                  class="dropdown-item"
                  data-rr-ui-dropdown-item=""
                  href="#"
                  role="button"
                  tabindex="0"
                >
                  High
                   
                </a>
                <span
                  class="bg-secondary-subtle text-secondary border text-center dropdown-item-text"
                >
                  Filter by criteria
                </span>
                <span
                  class="dropdown-item-text"
                >
                  <input
                    class="mt-2 mb-2 form-control"
                    placeholder="Search"
                    value=""
                  />
                </span>
                <span
                  class="bg-secondary-subtle text-secondary border text-center dropdown-item-text"
                >
                  Clear All Filters
                </span>
                <a
                  class="dropdown-item"
                  data-rr-ui-dropdown-item=""
                  href="#"
                  role="button"
                  tabindex="0"
                >
                  <button
                    class="bi bi-x-circle w-100 mt-1 btn btn-danger btn-sm"
                    type="button"
                  />
                </a>
              </div>
            </div>
            <div
              class="dropdown-menu-header dropdown btn-group"
              role="group"
            >
              <button
                aria-expanded="false"
                class="dropdown-toggle btn btn-light btn-sm"
                type="button"
              >
                <i
                  class="bi bi-grid-fill text-primary"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row bg-secondary-subtle p-2 justify-content-between align-items-center"
    >
      <div
        class="col-4 col-md-4 col-lg-2"
      >
        <h3
          class="m-0 text-center"
        >
          Kanban Board
        </h3>
      </div>
      <div
        class="col-4 col-md"
      >
        <p
          class="bg-white rounded text-center p-1 m-0 p-md-2 p-lg-1 bg-transparent text-secondary"
        >
          Completed 
          0
           task
          s
           of 
          0
           (
          0
          %)
        </p>
      </div>
      <div
        class="col-4 col-md-2 col-lg-1"
      >
        <div
          class="d-flex justify-content-evenly"
        >
          <div
            class="dropdown-menu-header show dropdown btn-group"
            role="group"
          >
            <button
              aria-expanded="true"
              class="dropdown-toggle show btn btn-light btn-sm"
              type="button"
            >
              <i
                class="bi bi-filter"
              />
            </button>
            <div
              aria-labelledby=""
              class="dropdown-menu show"
              x-placement="bottom-start"
            >
              <span
                class="bg-secondary-subtle text-secondary border text-center dropdown-item-text"
              >
                Filter by priority
              </span>
              <a
                class="dropdown-item"
                data-rr-ui-dropdown-item=""
                href="#"
                role="button"
                tabindex="0"
              >
                All
                 
              </a>
              <a
                class="dropdown-item"
                data-rr-ui-dropdown-item=""
                href="#"
                role="button"
                tabindex="0"
              >
                Low
                 
              </a>
              <a
                class="dropdown-item"
                data-rr-ui-dropdown-item=""
                href="#"
                role="button"
                tabindex="0"
              >
                Medium
                 
              </a>
              <a
                class="dropdown-item"
                data-rr-ui-dropdown-item=""
                href="#"
                role="button"
                tabindex="0"
              >
                High
                 
              </a>
              <span
                class="bg-secondary-subtle text-secondary border text-center dropdown-item-text"
              >
                Filter by criteria
              </span>
              <span
                class="dropdown-item-text"
              >
                <input
                  class="mt-2 mb-2 form-control"
                  placeholder="Search"
                  value=""
                />
              </span>
              <span
                class="bg-secondary-subtle text-secondary border text-center dropdown-item-text"
              >
                Clear All Filters
              </span>
              <a
                class="dropdown-item"
                data-rr-ui-dropdown-item=""
                href="#"
                role="button"
                tabindex="0"
              >
                <button
                  class="bi bi-x-circle w-100 mt-1 btn btn-danger btn-sm"
                  type="button"
                />
              </a>
            </div>
          </div>
          <div
            class="dropdown-menu-header dropdown btn-group"
            role="group"
          >
            <button
              aria-expanded="false"
              class="dropdown-toggle btn btn-light btn-sm"
              type="button"
            >
              <i
                class="bi bi-grid-fill text-primary"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
